examples/jsm/exporters/GLTFExporter.js

```diff
@@ -1203,27 +1267,41 @@ class GLTFWriter {
 	 */
 	processTexture( map ) {
 
+		const writer = this;
+		const options = writer.options;
 		const cache = this.cache;
 		const json = this.json;
 
 		if ( cache.textures.has( map ) ) return cache.textures.get( map );
 
 		if ( ! json.textures ) json.textures = [];
 
-		let mimeType = map.userData.mimeType;
+
+		let modifiedMap = map;
+
+		// make non-readable textures (e.g. CompressedTexture) readable by blitting them into a new texture
+		if ( typeof CompressedTexture !== 'undefined' && map instanceof CompressedTexture ) {
+
+			modifiedMap = decompress( map, options.maxTextureSize );
+			// remove from cache, as the underlaying canvas is always the same between decompressed textures
+			cache.images.delete( modifiedMap.image );
```

Comment:
I don't quite follow why the cache needs to be cleared. I believe we have this cache to avoid embedding the same texture data more than once. If the canvas is being shared, we may need to cache something else (`.getImageData()`?) instead.

Reply:
Yes, it's because the Canvas is shared here. I wasn't sure if caching getImageData instead would be too intrusive in this PR, can change that.

Reply:
If it's not too messy I'd go for it, yeah!